ambry-router/src/main/java/com.github.ambry.router/GetManager.java

```diff
@@ -38,29 +80,89 @@
    */
   GetManager(ClusterMap clusterMap, ResponseHandler responseHandler, RouterConfig routerConfig,
       NonBlockingRouterMetrics routerMetrics, Time time) {
-    //@todo
+    this.clusterMap = clusterMap;
+    this.responseHandler = responseHandler;
+    this.routerConfig = routerConfig;
+    this.routerMetrics = routerMetrics;
+    this.time = time;
+    getOperations = Collections.newSetFromMap(new ConcurrentHashMap<GetOperation, Boolean>());
   }
 
-  void submitGetBlobOperation(String blobId, FutureResult<ReadableStreamChannel> futureResult,
-      Callback<ReadableStreamChannel> callback) {
-    //@todo
+  /**
+   * Submit an operation to get the BlobInfo associated with a blob asynchronously.
+   * @param blobId the blobId for which the BlobInfo is being requested, in string form.
+   * @param futureResult the {@link FutureResult} that contains the pending result of the operation.
+   * @param callback the {@link Callback} object to be called on completion of the operation.
+   */
+  void submitGetBlobInfoOperation(String blobId, FutureResult<BlobInfo> futureResult, Callback<BlobInfo> callback) {
+    try {
+      GetBlobInfoOperation getBlobInfoOperation =
+          new GetBlobInfoOperation(routerConfig, clusterMap, responseHandler, blobId, futureResult, callback, time);
+      getOperations.add(getBlobInfoOperation);
+    } catch (RouterException e) {
+      NonBlockingRouter.completeOperation(futureResult, callback, null, e);
+    }
   }
 
-  void submitGetBlobInfoOperation(String blobId, FutureResult<BlobInfo> futureResult,
-      Callback<BlobInfo> callback) {
-    //@todo
+  /**
+   * Submit an operation to get a blob asynchronously.
+   * @param blobId the blobId for which the BlobInfo is being requested, in string form.
+   * @param futureResult the {@link FutureResult} that contains the pending result of the operation.
+   * @param callback the {@link Callback} object to be called on completion of the operation.
+   */
+  void submitGetBlobOperation(String blobId, FutureResult<ReadableStreamChannel> futureResult,
+      Callback<ReadableStreamChannel> callback) {
+    // @todo
   }
 
-  void poll(List<RequestInfo> requests) {
-    //@todo
+  /**
+   * Creates and returns requests in the form of {@link RequestInfo} to be sent to data nodes in order to complete
+   * get operations. Since this is the only method guaranteed to be called periodically by the RequestResponseHandler
+   * thread in the {@link NonBlockingRouter} ({@link #handleResponse} gets called only if a
```

Comment:
No. There is the start of the parenthesis between them. It should be clear in the generated doc. It says, "Since this is the only method guaranteed to be called periodically by the RequestResponseHandler thread in the NonBlockingRouter (handleResponse gets called only if a response is received for a get operation), any error handling or operation completion and cleanup also usually gets done in the context of this method."